### PR TITLE
Enforce container naming convention specified in appliance config

### PR DESCRIPTION
**Pull request by hickeng**
This is #6265 but applied to master. @cgtexmex contribued a chunk of this
code and will be picking it up from here.

Provides basic container naming convention via {id} and {name} tokens.
The naming is enforced only on vSphere and will not be visible via
the docker client.

Adds retry logic in rename call for ConflictError case

Missing:
1. tests
2. checks on the vSAN impact of changing the displayName like this
(might have to be non-vSAN only).
3. documentation

Towards: #6267

### CODE REVIEW

**Comment by chengwang86** [comment]
Overall LGTM. Just a couple of places where a little bit elaboration is needed:

1. why do we decide to only allow {name} or {id} but now allow for {name}-{id}? Is it because of the additional complexity or DBS restrictions/requirements?

2. why do we only use the template when the nameConvention is {name}? The case where the template has `{id}` is not quite straight-forward. If I do `docker rename newName` while the `--container-name-convention` is `MyOrg-{id}`. Then the `newName` will not be used if this PR is merged. Is this the reason why we obsolete `{id}`? https://github.com/vmware/vic/pull/6284/files#diff-68bcd1ace5fa1fc82a91d9d6852ae463R153

File: cmd/vic-machine/common/container.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	ContainerNameConvention string `cmd:"container-name-convention"`
+}
+
+func (c *ContainerConfig) ContainerFlags() []cli.Flag {
+	return []cli.Flag{
+		// container naming convention
+		cli.StringFlag{
+			Name:        "container-name-convention, cnc",
+			Value:       "",
+			Usage:       "Provide a naming convention. Allows a token of '{name}' or '{id}', that will be replaced.",
```
Comment (chengwang86) [inline cmd/vic-machine/common/container.go]: Can we make it clear to the user that this option only changes the `display name` of a containerVM on vsphere. So if `docker ps` will still show the pretty name of a containerVM as usual.

Comment (cgtexmex) [reply cmd/vic-machine/common/container.go]: Yes..I think that's needed, but let's do it in a future iteration.

**Comment by cgtexmex** [comment]
@chengwang86 here are my thoughts on your questions - @hickeng can add his POV:

1.  First to avoid complexity -- start simple and grow as needed.  Secondly, {name}-{id} is essentially what we have as the default naming convention. 
2.  In a rename of course the {name} changes, so we apply that to vSphere as well.  In the case of rename with {id} as the convention...nothing about the vSphere display name changes.  The user provided template is static and the ID is immutable, so there's nothing to change on vSphere.

HTH..

**Comment by cgtexmex** [comment]
LGTM

**Pull request merged**